src/libstd/sys/mod.rs

```diff
@@ -32,6 +32,10 @@
 
 pub use self::imp::*;
 
+#[cfg(target_os = "redox")]
+#[path = "redox/mod.rs"]
+mod imp;
```

Comment:
As we have a lot of different code, we create a new target family and a new `sys::redox` module